include/hbba_lite/core/Strategy.h

```diff
@@ -136,6 +138,10 @@ class FilterPool
 protected:
     virtual void applyEnabling(const std::string& name, const FilterConfiguration& configuration) = 0;
     virtual void applyDisabling(const std::string& name) = 0;
+
+    // Useful for decorators
+    void callApplyEnabling(FilterPool& filterPool, const std::string& name, const FilterConfiguration& configuration);
+    void callApplyDisabling(FilterPool& filterPool, const std::string& name);
```

Comment:
Revert the logic to use a Template Method pattern here too?

Reply:
Actually, don't think it's possible as decorators are not subclasses.
But, the `callApply` variants could probably be replaced by a call to the `apply` method on the decorated object directly?

Reply:
I don't think a template method would resolve the issue that I had.

In `RosLogFilterPoolDecorator::applyEnabling`, it is not possible to call `m_filterPool->applyEnabling` because applyEnabling is protected. I think the decorator pattern is the right way to implement the logging.